cohortextractor/backends/base.py

```diff
@@ -10,17 +16,13 @@ def register_backend(backend_class):
 
 
 class BaseBackend:
-    backend_id = NotImplemented
-    query_engine_class = NotImplemented
-    patient_join_column = NotImplemented
+    backend_id: str
+    query_engine_class: type["BaseSQLQueryEngine"]
```

Comment:
I think you can just use the class name here rather than a string, can't you? (And therefore probably do away with the `TYPE_CHECKING` guard on the import.

Reply:
I was trying to avoid potential cyclic imports but we're not hitting that yet so probably YAGNI.